Support blocking exception mappers in REST Client Reactive (#32644)
quarkusio/quarkus: docs/src/main/asciidoc/rest-client-reactive.adoc

NOTE: Methods annotated with `@ClientExceptionMapper` can also take a `java.lang.reflect.Method` parameter which is useful if the exception mapping code needs to know the REST Client method that was invoked and caused the exception mapping code to engage.

=== Using @Blocking annotation in exception mappers

> In scenarios to handle large files, the client might require to return directly the InputStream object:

Reviewer: ```suggestion
In cases that warrant using `InputStream` as the return type of REST Client method (such as when large amounts of data need to be read)
```

Author: updated

[source, java]
----
@Path("/echo")
@RegisterRestClient
public interface EchoClient {

    @GET
    InputStream get();
}
----

> This will work as expected, but if you try to read this InputStream object in a custom exception mapper, you will receive a BlockingNotAllowedException exception. This is because the REST Client Reactive uses the Event Loop thread executor by default which does not allow to perform IO operations. 

Reviewer: ```suggestions
This will work as expected, but if you try to read this InputStream object in a custom exception mapper, you will receive a `BlockingNotAllowedException` exception. This is because `ResponseExceptionMapper` classes are run on the Event Loop thread executor by default - which does not allow to perform IO operations.
```

Author: updated
